src/uu/cp/src/cp.rs

```diff
@@ -1089,7 +1089,7 @@ fn copy_attribute(source: &Path, dest: &Path, attribute: &Attribute) -> CopyResu
 }
 
 #[cfg(not(windows))]
-#[allow(clippy::unnecessary_wraps)] // needed for windows version
+#[allow(clippy::unnecessary_unwrap)] // needed for windows version
```

Comment:
Is this allow needed? Also, it seems like we're not handling errors in `symlink_file` for linux/macos, but always returning Ok(()). Maybe it should be the same as for windows below?

Reply:
Haven't checked yet (it's so cumbersome to boot up a windows VM).
I also noticed errors are not handled in `symlink_file`, however there's an 'unwrap' which should be enough, i.e. print the error message and then panic.

Reply:
I'm not sure I understand. Which `unwrap` are you referring to? Any error from the call to `std::os::unix::fs::symlink(source, dest).context(context)` is not handled, or am I missing something?

Also, you changed this from `unnecessary_wraps` to `unnecessary_unwrap`, but... there's no unwrap in here. That's what confused me. Anyways,I'm booting up my windows VM to check :)

Reply:
Removing this allow doesn't make clippy complain on windows (checked in a VM).
However, there are two other warnings:
```

warning: writing `&PathBuf` instead of `&Path` involves a new object where a slice will do
   --> src\uu\du\src/du.rs:150:27
    |
150 | fn get_size_on_disk(path: &PathBuf) -> u64 {
    |                           ^^^^^^^^ help: change this to: `&Path`
    |
    = note: `#[warn(clippy::ptr_arg)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg

warning: writing `&PathBuf` instead of `&Path` involves a new object where a slice will do
   --> src\uu\du\src/du.rs:182:24
    |
182 | fn get_file_info(path: &PathBuf) -> Option<FileInfo> {
    |                        ^^^^^^^^ help: change this to: `&Path`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_arg

warning: 2 warnings emitted
```
With stable there is another one:
```
warning: name `DWORD` contains a capitalized acronym
   --> src\uu\rm\src/rm.rs:406:14
    |
406 |     pub type DWORD = c_ulong;
    |              ^^^^^ help: consider making the acronym lowercase, except the initial letter: `Dword`
    |
    = note: `#[warn(clippy::upper_case_acronyms)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#upper_case_acronyms

warning: 1 warning emitted
```

Reply:
> I'm not sure I understand. Which unwrap are you referring to? Any error from the call to std::os::unix::fs::symlink(source, dest).context(context) is not handled, or am I missing something?

Oh, I was referring to L539:
https://github.com/uutils/coreutils/blob/3d42d41a72e7f43d6d5a88d73eed4dae181f7dc9/tests/common/util.rs#L533-L540

> Also, you changed this from unnecessary_wraps to unnecessary_unwrap, but... there's no unwrap in here. That's what confused me.

There is no `unnecessary_wraps`, probably a typo and clippy's suggestion was to use `unnecessary_unwrap`.